External/FEXCore/include/FEXCore/Utils/File.h

```diff
@@ -0,0 +1,253 @@
+#pragma once
+#include <FEXCore/fextl/allocator.h>
+#include <FEXCore/Utils/EnumOperators.h>
+
+#ifndef _WIN32
+#include <fcntl.h>
+#include <unistd.h>
+#else
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#undef ERROR
+#endif
+
+namespace FEXCore::File {
+  enum class FileModes : uint32_t {
+    READ     = (1U << 0),
+    WRITE    = (1U << 1),
+    CREATE   = (1U << 2),
+    TRUNCATE = (1U << 3),
+  };
+
+  enum class SeekOp {
+    BEGIN,
+    CURRENT,
+    END,
+  };
+
+  FEX_DEF_NUM_OPS(FileModes)
+
+  class File final {
+    public:
+#ifndef _WIN32
+      using FileHandleType = int;
+#else
+      using FileHandleType = HANDLE;
+#endif
+
+      File() = default;
+
+      File(const char *Filepath, FileModes Modes) {
+#ifndef _WIN32
+        auto Disp = TranslateModes(Modes);
+        Handle = open(Filepath, Disp, DEFAULT_USER_PERMS);
+        IsValidHandle = Handle != -1;
+#else
+        auto Disp = TranslateModes(Modes);
+        if (Disp.CreationFlag == OPEN_ALWAYS && Disp.TruncateOnExist) {
+          // If Open + Truncate then try to open with truncate behaviour first.
+          Handle = CreateFileA(Filepath, Disp.Access, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nullptr, TRUNCATE_EXISTING, FILE_ATTRIBUTE_NORMAL, nullptr);
```

Comment:
maybe pull ` FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE` into a const local so it isn't needed to be duplicated across 3 calls, given how long it is

Reply:
Converted in to a local constant